authors/apps/articles/views.py

```diff
@@ -275,3 +275,72 @@ def get(self, request):
         return Response(
             {'tags': msg},
             status=status.HTTP_404_NOT_FOUND)
+
+
+class BookmarkAPIView(generics.ListAPIView):
+    """
+    Class returns all articles in bookmarks of the user logged in
+    """
+    permission_classes = (permissions.IsAuthenticated,)
+    serializer_class = serializers.BookmarkSerializer
+    renderer_classes = (BookmarkJSONRenderer,)
+
+    def get_queryset(self):
+        """
+        This view should return a list of all the bookmarks
+        for the currently authenticated user.
+        """
+        user = self.request.user
+        return Bookmark.objects.filter(user=user)
+
+
+class ArticleBookmarkAPIView(generics.GenericAPIView):
+    """Class adds or deletes an article from users bookmarks"""
+    permission_classes = (permissions.IsAuthenticated,)
+    serializer_class = serializers.BookmarkSerializer
+
+    def post(self, request, slug):
+        """
+        Add article to bookmarks of the user
+        Checks to see article being added exists
+        Checks to see if article has already been added to boomarks
+        """
+        article = get_single_article_using_slug(slug)
+        if not article:
+            message = f'This article does not exist'
+            return Response({
+                'message': message},
+                status=status.HTTP_404_NOT_FOUND)
+        try:
+            Bookmark.objects.get(article=article, user=request.user)
+            return Response({
+                'message': f'Article already exists in bookmarks'
+            }, status=status.HTTP_400_BAD_REQUEST)
+        except Bookmark.DoesNotExist:
+            Bookmark.objects.create(article=article, user=request.user)
+            message = f'Article has been added to bookmarks'
+            return Response({
+                'message': message},
+                status=status.HTTP_201_CREATED)
+
+    def delete(self, request, slug):
+        """
+        Delete article from bookmarks of the user
+        Checks to see article being deleted exists
+        Checks to see if article has already been deleted from boomarks
+        """
+        article = get_single_article_using_slug(slug)
+        if not article:
+            return Response({
+                'message': 'This article does not exist'},
+                status=status.HTTP_404_NOT_FOUND)
+
+        bookmark = Bookmark.objects.filter(article=article, user=request.user)
+
+        if bookmark:
+            bookmark.delete()
+            message = f'Article has been deleted from your bookmarks'
+            return Response({'message': message})
```

Comment:
Great work at @missvicki just a few observations you're missing the status codes for this response

Reply:
Resolving this